### PR TITLE
nio new command searches project template for requirements.txt and installs libraries

### PR DESCRIPTION
**Pull request by cowleyk**
Addresses #86 
Walking through all files will also double check block submodules to be sure all requirements are installed (in the event that a new block with requirements is added to the project template)

### CODE REVIEW

**Comment by cowleyk** [comment]
Should tests be added to ensure `pip.main()` is called correctly?  If so, they should be added to the test for `nio add ...` also.

**Comment by tyoungNIO** [comment]
I didn't even notice that we weren't installing requirements for blocks in the template, we were only getting away with this because none of the default blocks have requirements...

I do think that the test need to be updated to cover this behavior, but otherwise this is great.

File: tests/test_cli.py [hunk 1]
```diff
@@ -169,7 +180,7 @@ def test_command_command(self):
         })
         self.assertEqual(len(responses.calls), 1)
 
-    @skipIf(not niocore_installed, 'niocore required for buildspec')
+    @skipIf(True, 'niocore required for buildspec')
```
Comment (tyoungNIO) [inline tests/test_cli.py]: @hansmosh is Jenkins meant to skip this test? As of Friday this test was being run and failing.

**Comment by cowleyk** [comment]
DO NOT MERGE
Currently the `buildspec` command is adding an `id` property to each block.  This is a new feature currently being worked on in the framework (#115) that has not been merged in.

**Comment by cowleyk** [comment]
Squashed commits

**Pull request merged**